src/H5Dscatgath.c

```diff
@@ -490,46 +502,74 @@ H5D__scatgath_read(const H5D_io_info_t *io_info, const H5D_dset_io_info_t *dset_
     for (smine_start = 0; smine_start < dset_info->nelmts; smine_start += smine_nelmts) {
         size_t n; /* Elements operated on */
 
-        /* Go figure out how many elements to read from the file */
         assert(H5S_SELECT_ITER_NELMTS(file_iter) == (dset_info->nelmts - smine_start));
-        smine_nelmts = (size_t)MIN(dset_info->type_info.request_nelmts, (dset_info->nelmts - smine_start));
+
+        /* Determine strip mine size. First check if we're doing in-place type conversion */
+        if (in_place_tconv) {
+            /* If this is not a selection I/O operation and there is a background buffer, we cannot exceed
+             * request_nelmts.  It could be part of a selection I/O operation if this read is used to fill in
+             * a nonexistent chunk */
+            if (dset_info->type_info.need_bkg &&
```

Comment:
I'm unsure of the differences between `H5T_BKG_YES` and `H5T_BKG_TEMP`. Should this be compared as `(H5T_BKG_YES == dset_info->type_info.need_bkg)` as below?

Reply:
Background buffer can be either not needed at all (need_bkg is 0), needed for conversion but the contents on entry don't matter (H5T_BKG_TEMP), or it needs to be initialized with the destination contents (H5T_BKG_YES). This code was put in to avoid running off the end of the background buffer during in-place type conversion. Normally in-place type conversion can use the whole width of the I/O since the type conversion buffer is the read buffer, but if the background buffer was allocated by the non-selection pathway (and is therefore limited in size) then we need to limit the strip mine size to the size of the background buffer.. Come to think of it the check for !H5D__SCATGATH_USE_CMPD_OPT_READ is redundant since if in_place_tconv is true, as is the condition to enter this code, that macro will always be false.

Reply:
So for this check it doesn't matter if the background buffer needs to be initialized or not, TEMP or YES should both trigger the limited strip mine size.